### PR TITLE
Fix panic when parsing conflict markers

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ### Fixed bugs
 
+ * Fixed panic when parsing conflict markers of a particular form.
```
Comment (ilyagr) [inline CHANGELOG.md:20]: I'd add the word "invalid" before conflict markers

File: lib/src/conflicts.rs [hunk 2]
```diff
@@ -473,7 +473,12 @@ fn parse_conflict_hunk(input: &[u8]) -> Merge<BString> {
         }
     }
 
-    Merge::from_removes_adds(removes, adds)
+    if adds.len() == removes.len() + 1 {
+        Merge::from_removes_adds(removes, adds)
+    } else {
+        // Doesn't look like a conflict
```
Comment (ilyagr) [inline lib/src/conflicts.rs:479]: I'd say something like "We cannot parse these conflict markers" or "Doesn't look like a valid conflict"

File: lib/tests/test_conflicts.rs [hunk 3]
```diff
@@ -748,6 +748,28 @@ fn test_parse_conflict_different_wrong_arity() {
     )
 }
 
+#[test]
+// TODO: Should *not* panic
+#[should_panic]
+fn test_parse_conflict_missing_removes() {
+    assert_eq!(
```
Comment (ilyagr) [inline lib/tests/test_conflicts.rs:753]: Assuming this is correct, I'd add a comment here:

> `jj` should never produce conflict markers like this, but we don't want to panic while parsing them.

**Review by ilyagr — APPROVED** [review]
Great, thank you!

(All of my comments are optional)

File: CHANGELOG.md (hunk 1 above)
Comment (martinvonz) [reply CHANGELOG.md]: Done

File: lib/src/conflicts.rs (hunk 2 above)
Comment (martinvonz) [reply lib/src/conflicts.rs]: Done (here and above)

**Pull request merged**